flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/FastDistanceVectorData.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.ml.linalg.DenseVector;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+
+/** Save the data for calculating distance fast. The FastDistanceMatrixData */
+public class FastDistanceVectorData implements Serializable, Cloneable {
+    /** Stores the vector(sparse or dense). */
+    final DenseVector vector;
+
+    /**
+     * Stores some extra info extracted from the vector. For example, if we want to save the L1 norm
+     * and L2 norm of the vector, then the two values are viewed as a two-dimension label vector.
+     */
+    public DenseVector label;
```

Comment:
It seems that `label` length is always 1. Could we make it a double for now?

If we need to make this a vector in the future, could you explain the use-case? It looks like whatever normalization setup (e.g. l1, l2) setup we use, there will be just one double value representation the normalization value of this vector?

And suppose we make this vector, if the length of this vector can be determined when `FastDistanceVectorData` is constructed, could this field be `public final`?

Reply:
This class has been deleted.